src/note/asset_base.rs

```diff
@@ -165,13 +165,13 @@ pub mod testing {
             isk in arb_issuance_authorizing_key(),
             str in "[A-Za-z]{255}"
         ) -> AssetBase {
-            AssetBase::derive(&IssuanceValidatingKey::from(&isk), &str)
+            AssetBase::derive(&IssuanceValidatingKey::from(&isk), &str.as_bytes().into())
         }
     }
 
     prop_compose! {
         /// Generate an asset ID using a specific description
-        pub fn zsa_asset_base(asset_desc: String)(
+        pub fn zsa_asset_base(asset_desc: Vec<u8>)(
```

Comment:
We are not consuming the `asset_desc` inside the function, so should be `&Vec<u8>`
Should follow this logic everywhere.

Reply:
Trying to make this change leads to the following error

```
hidden type for `impl proptest::strategy::Strategy<Value = AssetBase>` captures lifetime that does not appear in bounds [E0700] Help: to declare that the `impl Trait` captures `'_`, you can add an explicit `'_` lifetime bound
hidden type for `impl proptest::strategy::Strategy<Value = note::asset_base::AssetBase>` captures lifetime that does not appear in bounds [E0700] Help: to declare that the `impl Trait` captures `'_`, you can add an explicit `'_` lifetime bound
```

Leaving this open to figure a workaround, and resolving the other comments.